pants

```diff
@@ -34,10 +34,24 @@ fi
 
 PANTS_BOOTSTRAP="${PANTS_SETUP_CACHE}/bootstrap-$(uname -s)-$(uname -m)"
 
-VIRTUALENV_ZIPAPP=virtualenv.pyz
+PEX_VERSION=2.1.42
+PEX_URL="https://github.com/pantsbuild/pex/releases/download/v${PEX_VERSION}/pex"
+PEX_EXPECTED_SHA256="69d6b1b1009b00dd14a3a9f19b72cff818a713ca44b3186c9b12074b2a31e51f"
+
 VIRTUALENV_VERSION=20.4.7
-VIRTUALENV_ZIPAPP_URL="https://raw.githubusercontent.com/pypa/get-virtualenv/${VIRTUALENV_VERSION}/public/${VIRTUALENV_ZIPAPP}"
-VIRTUALENV_ZIPAPP_EXPECTED_SHA256="9d5c8b09cdeaf44414f07902d2ae45c4bb8a2085717d66f82ae389a9c6f94425"
+VIRTUALENV_REQUIREMENTS=$(
+cat << EOF
+virtualenv==${VIRTUALENV_VERSION} --hash sha256:2b0126166ea7c9c3661f5b8e06773d28f83322de7a3ff7d06f0aed18c9de6a76
```

Comment:
Alternatively, we could just build and host out own virtualenv.pex to retain single file download. As this PR stands, a shop with a private PyPI might have a lot more work to do to get this script working with their setup.

Reply:
Not a bad idea, although then that private PyPI shop would have to host that virtualenv.pex?

Reply:
Although I guess now they have to host pex.pex which has the same implications.

Reply:
Yeah, this is an existing problem with the virtualenv.pyz, the problem just flips to the Pex PEX. I was pointing out the problem expands a bit though, since PyPI (or some private index) gets involved in building the virtualenv.pex whereas it did not get involved before.